### PR TITLE
Changes to bm2rgbi tool

### PR DESCRIPTION
**Pull request by csatti**
Added option to change the byte order (big endian / little endian).
Added option to save the image as flipped.
Compressed data is also exported into a C file for easier integration into user project (xxd.exe).

### CODE REVIEW

**Comment by punkkeks** [comment]
Maybe you could use an enum instead of an bool for the byteorder?

**Comment by csatti** [comment]
I will do that.

**Comment by punkkeks** [comment]
You don't have to close your PR, just commit the fix into the branch and it will update it automaticaly

File: utils/bm2rgbi/bm2rgbi/Generic565Converter.cs [hunk 1]
```diff
@@ -5,100 +5,98 @@
...
+
     /// <summary>
-    /// generic converter for 565 format 64K colours
-    /// Format: RRRRRGGGGGGBBBBB (RGB)
-    /// Format: BBBBBGGGGGGRRRRR (BGR)
+    /// Convert to 565 format RGB
     /// </summary>
 
-    public class Generic565Converter
+    public void convertRGB(Bitmap bm, FileStream fs, Endianness ByteOrder)
```
Comment (punkkeks) [inline utils/bm2rgbi/bm2rgbi/Generic565Converter.cs]: why the change from class back to function?

**Comment by csatti** [comment]
The executables were included in the original tool.

File: utils/bm2rgbi/bm2rgbi/IBitmapConverter.cs [hunk 2]
```diff
@@ -3,9 +3,24 @@
...
-    void convert(Bitmap bm,FileStream fs, bool ByteOrder);
+    /// <summary>
+    /// Specifies the endianness of the target device
+    /// </summary>
+    public enum Endianness
+    {
+        /// <summary>
+        /// Most significant byte first
+        /// </summary>
+        BigEndian = 0,
```
Comment (punkkeks) [inline utils/bm2rgbi/bm2rgbi/IBitmapConverter.cs]: why the implicit = 0 and = 1? Do you really need to specify them?

Comment (csatti) [inline utils/bm2rgbi/bm2rgbi/IBitmapConverter.cs]: Not really.

**Comment by punkkeks** [comment]
Oh okay. Then it should be right ;)

File: utils/bm2rgbi/bm2rgbi/Generic565Converter.cs (hunk 1 above)
Comment (csatti) [inline utils/bm2rgbi/bm2rgbi/Generic565Converter.cs]: Hmm, must have been some auto refactor, I will check it.

Comment (csatti) [inline utils/bm2rgbi/bm2rgbi/Generic565Converter.cs]: It's not changed. Just the formatting is not consistent in the original source and git thinks I have changed most of the source...

Comment (punkkeks) [inline utils/bm2rgbi/bm2rgbi/Generic565Converter.cs]: sorry... yeah, I see it now. Sorry for the noise

**Comment by csatti** [comment]
I agree about the redundancy, but the original source also had a lot. I think the idea was to keep everything related to the conversion for a device and format at the same place for easier overview.

**Comment by punkkeks** [comment]
I did not say it's your fault :) It's just something to keep in mind for the future maybe.

**Comment by csatti** [comment]
Ok. :)
